services/productService.js

```diff
@@ -0,0 +1,46 @@
+const productDao = require('../models/productDao')
+const jwt = require('../middleware/jwt')
+
+const getProducts = async(brandName,scentName) =>{
+    const productAll = await productDao.getProducts(brandName,scentName)
+    return productAll
+}
+
+const getProduct = async(id) =>{
+    const product = await productDao.getProduct(id)
+    if(product.length === 0){
+        const err = new Error()
+        err.statusCode = 400
+        err.message = "없는 상품입니다."
+        throw err;
+    }
+    return product
+} 
+
+const createProduct = async(productName,price,description,brandId,scentId) =>{
+    const product = await productDao.createProduct(productName,price,description,brandId,scentId)
+    return product
+}
+
+const updateProduct = async(productId,productName,price,description,brandId,scentId)=>{
+    const getProduct = await productDao.getProduct(id)
+    const product = await productDao.updateProduct(productId,productName,price,description,brandId,scentId)
+    if(getProduct.length === 0){
+        const err = new Error()
+        err.statusCode = 400
+        err.message = "없는 상품입니다."
+        throw err;
+    }
+    return product 
+
```

Comment:
```suggestion
    const getProduct = await productDao.getProduct(id)
    
    if(getProduct.length === 0){
        const err = new Error()
        err.statusCode = 400
        err.message = "없는 상품입니다."
        throw err;
    }
    
    const product = await productDao.updateProduct(productId,productName,price,description,brandId,scentId)
    return product 

```
- product 업데이트를 하기 전에 product 존재 여부를 확인하는 로직이 먼저 오는 것이 맥락상 좋을 것 같습니다!